contracts/solidity/test/Transfer.sol

```diff
@@ -5,20 +5,61 @@ import "forge-std/Test.sol";
 import "../src/YABTransfer.sol";
 import {ERC1967Proxy} from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
 
-
 contract TransferTest is Test {
+    address public deployer = address(0xB321099cf86D9BB913b891441B014c03a6CcFc54);
+    address public marketMaker;
+
     YABTransfer public yab;
+    ERC1967Proxy public proxy;
+    YABTransfer public yab_caller;
 
     function setUp() public {
+        vm.startPrank(deployer);
+
         address snMessagingAddress = 0xde29d060D45901Fb19ED6C6e959EB22d8626708e;
         uint256 snEscrowAddress = 0x0;
         uint256 snEscrowWithdrawSelector = 0x15511cc3694f64379908437d6d64458dc76d02482052bfb8a5b33a72c054c77;
+        marketMaker = vm.envAddress("MM_ETHEREUM_WALLET");
+        
         yab = new YABTransfer();
-        ERC1967Proxy proxy = new ERC1967Proxy(address(yab), "");
-        YABTransfer(address(proxy)).initialize(snMessagingAddress, snEscrowAddress, snEscrowWithdrawSelector);
+        proxy = new ERC1967Proxy(address(yab), "");
+        yab_caller = YABTransfer(address(proxy));
+        yab_caller.initialize(snMessagingAddress, snEscrowAddress, snEscrowWithdrawSelector, marketMaker);
+
+        vm.stopPrank();
+    }
+
+    function testTransfer_mm() public { //ok
+        hoax(marketMaker, 100 wei);
+        yab_caller.transfer{value: 100}(1, 0x1, 100);
+    }
+
+    function testTransfer_fail() public { //ok
+        hoax(address(0x0000000000000000000000000000000000000001), 100 wei);
+        vm.expectRevert("Only Owner or MM can call this function");
+        yab_caller.transfer{value: 100}(1, 0x1, 100);
+    }
+
+    // function testWithdraw_mm() public { //not working
+    //     // escrow: 0x00d3d7c86ba3931b120dfb08a41f6b8e78e37128bf09eca76b6a639965e014d6
+    //     vm.prank(deployer);
+    //     yab_caller.setEscrowAddress(0x00d3d7c86ba3931b120dfb08a41f6b8e78e37128bf09eca76b6a639965e014d6);
+
+    //     hoax(marketMaker, 100 wei);
+    //     yab_caller.transfer{value: 100}(1, 0x1, 100);
+    //     hoax(marketMaker, 100 wei);
+    //     yab_caller.withdraw{value: 100}(1, 0x1, 100);
+    // }
```

Comment:
why is this commented? shouldn't this be working?

Reply:
To finish this test the L1-L2 messaging has to be mocked, and as it was said this was not an urgent test, i commented out the WIP test.
I could remove this test from this PR, or finish it (I would prefer finishing it)

Reply:
It would be great if you could add a way to mock this contract